### PR TITLE
Server side to client side data sync for diagnosis

### PR DESCRIPTION
**Pull request by mithun-grmtech**

### CODE REVIEW

File: webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ export default class clientSideTblMasterDiagnosis extends clientSideTableManage
 
       masterDiagnosisId: this.uid(() => intUniqueId()),
       diagnosisName: this.string(null),
+      isAssignedToPatient: this.boolean(false),
```
Comment (vikaskedia) [inline webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js:21]: Why is this needed?

File: webclient/components/1time-Mrow-mField/diagnosis/db/client-side/initialize-on-first-load.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <div>
+    <initializeDiagnosisComponent />
+  </div>
+</template>
+
+<script>
+import initializeDiagnosisComponent from '@/components/1time-Mrow-mField/diagnosis/db/client-side/static-data/insert-into-master-of-search-phrases-ct'
```
Comment (vikaskedia) [inline webclient/components/1time-Mrow-mField/diagnosis/db/client-side/initialize-on-first-load.vue:8]: Look at this has been divided into 3 steps by service statement component.

File: webclient/components/1time-Mrow-mField/diagnosis/edit-layer/add-ct.vue [hunk 3]
```diff
@@ -0,0 +1,298 @@
+<!-- Reference implementation -->
```
Comment (vikaskedia) [inline webclient/components/1time-Mrow-mField/diagnosis/edit-layer/add-ct.vue:1]: Why is there a seperate add component. The design of SS where the same ct can do add/remove seems to be better. Look at how the search at top of SS ct can be used to filter and narrow down.

Comment (mithun-grmtech) [reply webclient/components/1time-Mrow-mField/diagnosis/edit-layer/add-ct.vue:1]: Sir, I made the add component flown by the Remainder component.
okay, now I will change the diagnosis component flowed by the SS component.

File: webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js (hunk 1 above)
Comment (mithun-grmtech) [reply webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js:21]: Sir, 'isAssignedToPatient' field use for when the diagnosis assigned to a patient then the diagnosis remove from the master list of diagnosis.

File: servers/lumen/app/Http/Controllers/DiagnosisController.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+use Illuminate\Http\Request;
+use Ramsey\Uuid\Uuid;
+use DB;
+use Predis\Autoloader;
+
+class DiagnosisController extends Controller
+{
+    public function getAllTemporalDiagnosis()
+    {
+
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/DiagnosisController.php:15]: DB name should be sc_diagnosis

File: webclient/components/1time-Mrow-mField/diagnosis/db/client-side/initialize-on-first-load.vue [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <div>
+    <step1_init_search_phrases />
+    <step2_init_ss_field_names />
```
Comment (vikaskedia) [inline webclient/components/1time-Mrow-mField/diagnosis/db/client-side/initialize-on-first-load.vue:4]: Why does it say "ss_field_names" ?

File: webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js [hunk 6]
```diff
@@ -18,6 +18,7 @@ export default class clientSideTblMasterDiagnosis extends clientSideTableManage
 
       masterDiagnosisId: this.uid(() => intUniqueId()),
       diagnosisName: this.string(null),
+      // 'isAssignedToPatient' field use for when the diagnosis assigned to a patient then the diagnosis remove from the master list of diagnosis.
```
Comment (vikaskedia) [inline webclient/components/1time-Mrow-mField/diagnosis/db/client-side/structure/master-table-of-diagnosis.js:21]: this does not make sense

**Review by vikaskedia — CHANGES_REQUESTED** [review]
